src/category_theory/idempotents/biproducts.lean

```diff
@@ -0,0 +1,170 @@
+/-
+Copyright (c) 2022 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+
+import category_theory.idempotents.karoubi
+import category_theory.additive.basic
+
+/-!
+
+# Biproducts in the idempotent completion of a preadditive category
+
+In this file, we define an instance expressing that if `C` is an additive category,
+then `karoubi C` is also an additive category.
+
+We also obtain that for all `P : karoubi C` where `C` is a preadditive category `C`, there
+is a canonical isomorphism `P ⊞ P.supplement ≅ (to_karoubi C).obj P.X` in the category
+`karoubi C` where `P.supplement` is the formal direct factor of `P.X` corresponding to
+the idempotent endomorphism `𝟙 P.X - P.p`.
+
+-/
+
+noncomputable theory
+
+open category_theory.category
+open category_theory.limits
+open category_theory.preadditive
+
+universes v
+
+namespace category_theory
+
+namespace idempotents
+
+namespace karoubi
+
+variables {C : Type*} [category.{v} C] [preadditive C]
+
+namespace biproducts
+
+/-- The `bicone` used in order to obtain the existence of
+the biproduct of a functor `J ⥤ karoubi C` when `C`. -/
+@[simps]
+def bicone [has_finite_biproducts C] {J : Type v} [decidable_eq J] [fintype J]
+  (F : J → karoubi C) : bicone F :=
+{ X :=
+  { X := biproduct (λ j, (F j).X),
+    p := biproduct.map (λ j, (F j).p),
+    idempotence := begin
```

Comment:
Maybe this ship has sailed, but `idempotent` would have been a better name than `idempotence` here: it's too hard in English to consistently come up with variants of words that mean "the property of being X", and it's easier to consistently just say "X".

Reply:
What about the shorter `idem`? (Similarly as `factori[sz]ation` is often shortened as `fac` in other places in mathlib.) May I do such a change slightly later in a separate PR?

Reply:
Sure, either `idempotent` or `idem` are good as far as I'm concerned. Yes -- no need to worry about it for this PR, but it would be nice if we eventually get to it.